modules/processor/src/main/java/uk/gov/hmcts/reform/migration/ccd/CoreCaseDataService.java

```diff
@@ -38,34 +33,35 @@ public CaseDetails fetchOne(String authorisation, String caseId) {
         return coreCaseDataApi.getCase(authorisation, authTokenGenerator.generate(), caseId);
     }
 
-    public List<CaseDetails> fetchAll(String authorisation, String userId) {
-        int numberOfPages = getNumberOfPages(authorisation, userId, new HashMap<>());
+    public List<CaseDetails> fetchAll(String authorisation, String userId, String caseType) {
+        int numberOfPages = getNumberOfPages(authorisation, userId, caseType, new HashMap<>());
         return IntStream.rangeClosed(1, numberOfPages).boxed()
-            .flatMap(pageNumber -> fetchPage(authorisation, userId, pageNumber).stream())
+            .flatMap(pageNumber -> fetchPage(authorisation, userId, caseType, pageNumber).stream())
             .collect(Collectors.toList());
     }
 
-    private int getNumberOfPages(String authorisation, String userId, Map<String, String> searchCriteria) {
+    private int getNumberOfPages(String authorisation, String userId,  String caseType, Map<String, String> searchCriteria) {
         PaginatedSearchMetadata metadata = coreCaseDataApi.getPaginationInfoForSearchForCaseworkers(authorisation,
-            authTokenGenerator.generate(), userId, jurisdiction, caseType, searchCriteria);
+            authTokenGenerator.generate(), userId, null, caseType, searchCriteria);
         return metadata.getTotalPagesCount();
     }
 
-    private List<CaseDetails> fetchPage(String authorisation, String userId, int pageNumber) {
+    private List<CaseDetails> fetchPage(String authorisation, String userId,  String caseType, int pageNumber) {
         Map<String, String> searchCriteria = new HashMap<>();
         searchCriteria.put("page", String.valueOf(pageNumber));
-        return coreCaseDataApi.searchForCaseworker(authorisation, authTokenGenerator.generate(), userId, jurisdiction,
+        return coreCaseDataApi.searchForCaseworker(authorisation, authTokenGenerator.generate(), userId, null,
             caseType, searchCriteria);
     }
 
-    public CaseDetails update(String authorisation, String caseId, String eventId, String eventSummary, String eventDescription, Object data) {
+    public CaseDetails update(String authorisation, String eventId, String eventSummary, String eventDescription, String caseType, CaseDetails caseDetails) {
+        String caseId = String.valueOf(caseDetails.getId());
         UserDetails userDetails = idamClient.getUserDetails(AuthUtil.getBearerToken(authorisation));
 
         StartEventResponse startEventResponse = coreCaseDataApi.startEventForCaseWorker(
             AuthUtil.getBearerToken(authorisation),
             authTokenGenerator.generate(),
             userDetails.getId(),
-            jurisdiction,
+            null,
```

Comment:
I think jurisdiction is required for this API - but we should have it available in `caseDetails` so can get it from there. (Could also get case type from the `caseDetails`?)

Reply:
done